guide: Using Directus CMS with Neon Postgres and Astro to build a blog (#2693)
neondatabase/website: content/guides/directus-cms.md

---
author: rishi-raj-jain
enableTableOfContents: true
createdAt: '2024-12-22T00:00:00.000Z'
updatedOn: '2024-12-22T00:00:00.000Z'
title: Using Directus CMS with Neon Postgres and Astro to build a blog
subtitle: A step-by-step guide for building your own blog in an Astro application with Directus CMS and Postgres powered by Neon
---

In this guide, you will learn how to set up a serverless Postgres database with Neon, configure Directus CMS with Postgres, define a blog schema, and author content using Directus CMS. The guide also covers configuring API read permissions and building a dynamic frontend with Astro to display blog pages fetched from the Directus CMS instance.

## Prerequisites

To follow the steps in this guide, you will need the following:

- [Node.js 18](https://nodejs.org/en) or later
- A [Neon](https://console.neon.tech/signup) account
- [Docker Desktop](https://www.docker.com/products/docker-desktop/) set up locally

## Provisioning a serverless Postgres database

Using a serverless Postgres database powered by Neon lets you scale compute resources down to zero, which helps you save on compute costs.

> To get started, go to the [Neon console](https://console.neon.tech/app/projects) and create a project. You will then be presented with a dialog that provides a connection string of your database. You will be using the connection string to connect Directus CMS instance to your Postgres database.

Reviewer: ```suggestion
To get started, go to the [Neon console](https://console.neon.tech/app/projects) and create a project. You will then be presented with a dialog that provides a connection string for your database. You will be using the connection string to connect the Directus CMS instance to your Postgres database.
```

## Setting up Directus locally with Postgres

Let's begin with creating a Directus CMS backend to serve the content for blog posts. Open your terminal and run the following command:

```bash
mkdir directus-cms
cd directus-cms
```

and create a `docker-compose.yml` with the following code:

```yml
services:
  directus:
    image: directus/directus:11.3.5
    ports:
      - 8055:8055
    volumes:
      - ./database:/directus/database
      - ./uploads:/directus/uploads
      - ./extensions:/directus/extensions
    environment:
      SECRET: 'replace-with-secure-random-value'
      ADMIN_EMAIL: 'admin@example.com'
      ADMIN_PASSWORD: 'd1r3ctu5'
      DB_CLIENT: 'pg'
      DB_CONNECTION_STRING: 'postgresql://neondb_owner:...@ep-...us-east-1.aws.neon.tech/neondb?sslmode=require'
      DB_SSL__REJECT_UNAUTHORIZED: 'true'
      WEBSOCKETS_ENABLED: 'true'
      CORS_ENABLED: 'true'
      CORS_ORIGIN: '*'
```

> Now, update the `DB_CONNECTION_STRING` value to be the connection string obtained in the step earlier. Finally, run the following command to start the local Directus CMS instance:

Reviewer: ```suggestion
Now, set the `DB_CONNECTION_STRING` value to the connection string you obtained previously. Finally, run the following command to start the local Directus CMS instance:
```

```shell
docker-compose up -d
```

> Once the migrations are ran succesfully, the Directus CMS instance would be accessible at [localhost:8055/admin](http://localhost:8055/admin/). Sign in with the credentials set in the `docker-compose.yml` file.

Reviewer: ```suggestion
Once the migrations are run successfully, the Directus CMS instance will be accessible at [localhost:8055/admin](http://localhost:8055/admin/). Sign in with the credentials set in the `docker-compose.yml` file.
```

## Configure the authors schema in Directus CMS

> Navigate to the [Data Model](http://localhost:8055/admin/settings/data-model) view, and click the `+` icon to start creating a new data model.

Reviewer: ```suggestion
Navigate to the [Data Model](http://localhost:8055/admin/settings/data-model) view, and click the `+` icon to create a new data model.
```

> Set the **Name** field to be **authors**, press the next arrow icon, and save by clicking the tick icon.

Reviewer: ```suggestion
Set the **Name** field to **authors**, press the next arrow icon, and save by clicking the tick icon.
```

![](/guides/images/directus-cms/author-1.png)

Once that's done, click on `Create Field` to add a field to the `authors` schema.

![](/guides/images/directus-cms/author-2.png)

Select the field type as `Input` and set the `Key` as **name** to signify the field to be representative of author's name.

![](/guides/images/directus-cms/author-3.png)

Finally, click `Save` to finish adding the `name` field to the `authors` schema.

![](/guides/images/directus-cms/author-4.png)

> Now, let's move on to creating the post schema in Directus CMS.

Reviewer: ```suggestion
Now, let's move on to creating the `post` schema in Directus CMS.
```

## Configure the posts schema in Directus CMS

Navigate to the [Data Model](http://localhost:8055/admin/settings/data-model) view, and click the `+` icon to start creating a new data model.

> Set the **Name** field to be **posts**, press the next arrow icon, and save by clicking the `+` icon.

Reviewer: ```suggestion
Enter **posts** in the **Name** field, press the next arrow icon, and save by clicking the `+` icon.
```

> Then, follow the process as earlier to add the following fields to the `posts` schema:

Reviewer: ```suggestion
Then, follow the same process as earlier to add the following fields to the `posts` schema:
```
